subscribe.go

```diff
@@ -0,0 +1,71 @@
+package bps
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"sync"
+)
+
+var (
+	subReg   = map[string]SubscriberFactory{}
+	subRegMu sync.Mutex
+)
+
+// SubscribeOptions defines subscribe options.
+type SubscribeOptions struct {
+	// BatchSize holds desired message batch size.
+	// Default: 1.
+	BatchSize int
+}
+
+// ----------------------------------------------------------------------------
+
+// Subscriber defines the main subscriber interface.
+type Subscriber interface {
+	// Subscribe subscribes for topic messages and blocks till context is cancelled or error occurs.
+	//
+	// Handler is provided an atomic message batch (entire batch either succeeds or fails).
+	// Batch is guaranteed to contain at least 1 message.
+	Subscribe(
+		ctx context.Context,
+		topic string,
+		handler func([]Message) error,
+		opts *SubscribeOptions,
```

Comment:
Happy to turn it into `opts ...func(*SubscribeOptions)` for neat(er) API, should I though?